tests/unit/stream_alert_rule_processor/test_threat_intel.py

```diff
@@ -271,25 +289,27 @@ def test_segment(self):
         assert_equal(len(result[2]), 100)
         assert_equal(len(result[3]), 45)
 
-    @patch('botocore.client.BaseClient._make_api_call')
-    def test_query(self, mock_dynamodb):
+    @patch('boto3.client')
+    def test_query(self, mock_client):
         """Threat Intel - Test DynamoDB query method with batch_get_item"""
-        mock_dynamodb.return_value = MockDynamoDBClient.response()
+        threat_intel = StreamThreatIntel.load_from_config(self.config)
+        mock_client('dynamodb').batch_get_item.return_value = MockDynamoDBClient.response()
 
         test_values = ['1.1.1.2', '2.2.2.2', 'evil.com', 'abcdef0123456789']
-        result = self.threat_intel._query(test_values)
+        result = threat_intel._query(test_values)
         assert_equal(len(result), 2)
         assert_equal(result[0], {'ioc_value': '1.1.1.2', 'sub_type': 'mal_ip'})
         assert_equal(result[1], {'ioc_value': 'evil.com', 'sub_type': 'c2_domain'})
 
     @raises(ClientError)
-    @patch('botocore.client.BaseClient._make_api_call')
-    def test_query_with_exception(self, mock_dynamodb):
+    @patch('boto3.client')
+    def test_query_with_exception(self, mock_client):
         """Threat Intel - Test DynamoDB query method with exception"""
-        mock_dynamodb.return_value = MockDynamoDBClient.response(exception=True)
+        mock_client('dynamodb').batch_get_item.return_value = \
+            MockDynamoDBClient.response(exception=True)
 
         self.threat_intel._query(['1.1.1.2'])
-        assert_equal(mock_dynamodb.call_count, 3)
+        assert_equal(mock_client.call_count, 3)
```

Comment:
hey @chunyong-lin  my only comment is related to this test - I actually don't think this assert is ever occurring and this test will not be hit. Since the `threat_intel._query(['1.1.1.2'])` call is what raises the exception, this line will never be touched. You can verify this by adding a `print ...` before this assertion.

Reply:
Hi @ryandeivert You are absolutely right! No, this assertion doesn't take effect 😞 
Thank you for flag this out!